src/controllers/userController.ts

```diff
@@ -0,0 +1,57 @@
+import User from "@/models/userModel";
+import { Request, Response } from "express";
+import jwt from "jsonwebtoken";
+import { ObjectId } from "mongoose";
+import expressAsyncHandler from "express-async-handler";
+
+export const registerUser = expressAsyncHandler(
+  async (req: Request, res: Response) => {
+    const { name, email, password } = req.body;
+
+    const user: any = await User.create({
```

Comment:
what if user already exists?

Reply:
@alitjanjua it will throw error as there is a unique check placed on email in userSchema